### ISSUE TITLE
Consider applying the robustness principle with regard to user agent request validation

### ISSUE THREAD

**Issue by tplooker** [WICG/digital-identities#100]
For those unfamiliar with [this principle](https://en.wikipedia.org/wiki/Robustness_principle#:~:text=In%20computing%2C%20the%20robustness%20principle,what%20you%20accept%20from%20others%22.), it boils down to

> be conservative in what you do, be liberal in what you accept from others". It is often reworded as: "be conservative in what you send, be liberal in what you accept"

With regard to this API I would suggest the user agent / browser apply this philosophy when considering whether to, or how to perform request validation, to the extent it makes sense.

For example it may make sense for the browser to validate parts of a request to ensure it is well formed syntax wise, which might have certain developer ergonomic benefits and or security benefits, I do think this needs to be carefully analysed though and not just assumed the whole request must be strictly validated. Furthermore, I think it would be a mistake and would severely limit future extensibility if the user agent were to strictly validate the request, that is, in the event the browser receives a request which contains elements it doesn't understand, it shouldn't error, fail to forward the request to a wallet or only forward the request to the wallet with known request elements.

**Comment by RByers** [comment]
Agreed, and it's been part of Chrome plans from the start, though framed as [enabling competition and choice in credential formats](https://docs.google.com/presentation/d/1mz7AUJLc9y7zOU7CfLCsBSt3dGIkN_x61tpoRaQCQko/edit?resourcekey=0-NhvH5WIUclnZZPyRSyyLqQ#slide=id.g27f155581c9_0_10).  I've tried to capture this more explicitly in a new [explainer](https://github.com/WICG/digital-identities/blob/main/explainer.md) with:
> By separating the act of requesting from the specific protocol, we can enable flexibility and adaptability in both the protocol and credential formats. This way, the pace of changes in browsers won't hinder progress or block new developments.

Feedback / suggestions on better wording appreciated.

**Comment by hlflanagan** [comment]
Suggest reviewing https://datatracker.ietf.org/doc/rfc9413/ to see how it might apply to this issue and the spec text.

**Comment by marcoscaceres** [comment]
On the web platform, we've generally move away from the robustness principle towards being stricter, as it provides better security guarantees.

Specifically for this spec, when validation and processing of the format applies is really up to the protocol and associated spec, except for:

* when making a request, it MUST be JSON serializable.
* If required by the format, then the request must be validated - but the validation can happen outside the user agent (but might be in the content-process thread, for security reasons).   
* when receiving a response, it MUST be a JSON object in order for the user agent to construct a `DigitalCredential` object because `data` is defined as `[SameObject](https://webidl.spec.whatwg.org/#SameObject)] readonly attribute [object](https://webidl.spec.whatwg.org/#idl-object) [data](https://www.w3.org/TR/digital-credentials/#dom-digitalcredential-data);`. 

Thus, I think the only actionable thing here is just to make sure we specify that the `data` for `DigitalCredential` is JSON-parsed (if it's a string when received) and it's a JS Object.

**Comment by timcappalli** [comment]
This has been addressed in the spec!
